### ISSUE TITLE
Another way of generation of the right mask in treelstm.

### ISSUE THREAD

**Issue by haichao592** [jihunchoi/unsupervised-treelstm#4]
At first, I was a little bit confused by the code, as follow, in line 125-128 of treelstm.py:
`right_mask_leftmost_col = Variable(select_mask_cumsum.data.new(new_h.size(0), 1).zero_())`
`right_mask = torch.cat([right_mask_leftmost_col, select_mask_cumsum[:, :-1]], dim=1)`
When I figured out the meaning of that code snapet, I think here is another way that can produce the same mask in a more direct way:
`right_mask = select_mask_cumsum - select_mask`

And I want to know that whether they will output the same result in forward computation and receive the same gradients in backward propagation.
Would you please help me to verify the equivalence? 
Thank you!

**Comment by jihunchoi** [comment]
Hi, thanks for the suggestion!
I think the result should be the same after modifying the logic as you suggested.
I'll run a quick test soon, or it'll be great if you kindly test it and send a PR.
